packages/indexer-client/src/client/v8/operations.ts

```diff
@@ -1,28 +1,29 @@
-import type { Client } from 'es8';
-import type { BulkOperationType, BulkResponseItem } from 'es8/lib/api/types.js';
+import type { Client, estypes } from 'es8';
 
 import { type DataRecordNested, FailureData, IndexResult, logger } from '@overture-stack/maestro-common';
 
 export const bulkUpsert = async (client: Client, index: string, dataSet: DataRecordNested[]) => {
 	try {
-		const body = dataSet.flatMap((doc) => [{ index: { _index: index, _id: doc?.['id'] } }, doc]);
+		const body = dataSet.flatMap(({ _id, ...doc }) => [{ index: { _index: index, _id } }, doc]);
 
 		const response = await client.bulk({ refresh: true, body });
 
-		logger.debug(`Bulk upsert in index:'${index}'`, `# of documents: ${response.items.length}`);
+		logger.info(`Bulk upsert in index:'${index}'`, `# of documents: ${response.items.length}`);
 
 		const failureData: FailureData = {};
 
 		if (response.errors) {
 			// The items array has the same order of the dataset we just indexed.
 			// The presence of the `error` key indicates that the operation
 			// that we did for the document has failed.
-			response.items.forEach((item: Partial<Record<BulkOperationType, BulkResponseItem>>, indexItem: number) => {
-				const operation = item.index;
-				if (operation && 'error' in operation) {
-					failureData[indexItem] = [operation.error?.reason || 'error'];
-				}
-			});
+			response.items.forEach(
+				(item: Partial<Record<estypes.BulkOperationType, estypes.BulkResponseItem>>, indexItem: number) => {
+					const operation = item.index;
+					if (operation && 'error' in operation) {
+						failureData[indexItem] = [operation.error?.reason || 'error'];
```

Comment:
adding this to ensure clarity that it's logged as such on purpose

```suggestion
						failureData[indexItem] = [operation.error?.reason || 'unspecified error'];
```